src/elm/interval.coffee

```diff
@@ -235,18 +235,18 @@ module.exports.Collapse = class Collapse extends Expression
         if typeof b.low?.durationBetween == 'function'
           # handle DateTimes using durationBetween
           if a.high?.after b.low # overlap
-            a.high = b.high if !b.high? or b.high.after a.high
+            a.high = b.high if (!b.high? and b.low.before a.high) or b.high.after a.high
```

Comment:
Not sure that `and b.low.before a.high` is needed since you only get into this statement if `a.high?.after b.low`.

Reply:
Also, should that initial statement really be:
```coffee
if a.high?.sameOrAfter b.low # overlap
```
Technically if they are the same it's still considered an overlap (and should collapse).